modules/java-records-tests/src/integrationTest/java/org/apache/ignite/internal/schema/marshaller/ItTableViewTest.java

```diff
@@ -0,0 +1,165 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.schema.marshaller;
+
+import static org.apache.ignite.internal.lang.IgniteStringFormatter.format;
+import static org.apache.ignite.internal.schema.marshaller.AssertMarshaller.assertView;
+import static org.apache.ignite.internal.schema.marshaller.AssertMarshaller.assertViewThrows;
+import static org.junit.jupiter.api.Named.named;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.stream.Stream;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.internal.ClusterPerClassIntegrationTest;
+import org.apache.ignite.internal.schema.marshaller.Records.ComponentsExact;
+import org.apache.ignite.internal.schema.marshaller.Records.ComponentsNarrow;
+import org.apache.ignite.internal.schema.marshaller.Records.ComponentsReordered;
+import org.apache.ignite.internal.schema.marshaller.Records.ComponentsWide;
+import org.apache.ignite.internal.schema.marshaller.Records.ComponentsWrongTypes;
+import org.apache.ignite.internal.schema.marshaller.Records.NotAnnotatedNotMapped;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.lang.MarshallerException;
+import org.apache.ignite.table.Table;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Ensures that records and classes behave the same way.
+ */
+class ItTableViewTest extends ClusterPerClassIntegrationTest {
+    private static final String TBL = "TBL";
+    private static IgniteClient CLIENT;
+    private static Table TABLE_EMBEDDED;
+    private static Table TABLE_CLIENT;
+
+    private static Stream<Arguments> tableImpls() {
+        return Stream.of(
+                arguments(named("embedded", TABLE_EMBEDDED)),
+                arguments(named("client", TABLE_CLIENT))
+        );
+    }
+
+    @BeforeAll
+    static void beforeAll() {
+        CLIENT = IgniteClient.builder()
+                .addresses("localhost:10800")
+                .build();
+
+        sql(format(Records.SQL_PATTERN, TBL));
+
+        TABLE_EMBEDDED = CLUSTER.aliveNode().tables().table(TBL);
+        TABLE_CLIENT = CLIENT.tables().table(TBL);
```

Comment:
We need to find a way to synchronize access - a table created from one client (embedded in this case) doesn't guaranteed to be available on the other client immediately.
Simplest way could be to subclass this test so that each class will use the same client.